### PR TITLE
Feature/add product

### PR DESCRIPTION
**Pull request by AlessandroMarc**
## Describe your changes 

- Feature 1
- Feature 2
- ...
 
## Checklist before requesting a review

Put an "X" in the boxes that apply.

- [] I have performed a self-review of my code
- [] Does not contain commented code
 
## Screenshots (if appropriate)

| Screenshot 1 | Screenshot 2 |
| ------ | ------ |
| Place here | Place here |

### CODE REVIEW

**Comment by denise-techlead** [comment]
Great! Please wait a moment while I review your code. 👀

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

File: src/controllers/productController.js [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
+    } else {
+        res.status(500).json({ error });
+    }
 };
 
-const addProductEndpoint = async(req, res) => {
-    // TODO: Implement method
+const addProductEndpoint = async (req, res) => {
+    const { description, full_description, category, price, restaurant_id, active } = req.body;
+    const { success, productId, error } = await addProduct(id, description, full_description, category, price, restaurant_id, active);
```
Comment (leticiasassaki) [inline src/controllers/productController.js]: Hi! In this case, we don't need to use the id.
```suggestion
    const { success, productId, error } = await addProduct(description, full_description, category, price, restaurant_id, active);
```

File: src/controllers/productController.js [hunk 2]
```diff
@@ -1,14 +1,52 @@
...
+    if (success) {
+        res.status(200).json(productId);
```
Comment (leticiasassaki) [inline src/controllers/productController.js]: For this endpoint, we are expecting the status 201
```suggestion
        res.status(201).json(productId);
```

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Review by leticiasassaki — COMMENTED** [review]
Great job! I add just two points to improve the code!

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Comment by denise-techlead** [comment]
Great job on your pull request! Here's an explanation of how the code works:

- The code is written in JavaScript and runs on the Node.js platform.
- The `productController.js` file contains two endpoint functions: `getProductsEndpoint` and `addProductEndpoint`.
- The `getProductsEndpoint` function retrieves a list of products and returns them as a JSON response.
- The `addProductEndpoint` function accepts product data in the request body and adds a new product to the database.
- The `productRouter.js` file sets up the routes for the product endpoints ("/products" for GET and POST requests).
- The `productService.js` file contains functions for interacting with the database.
- The `getProducts` function executes a SELECT query to retrieve all products from the "Products" table.
- The `addProduct` function executes an INSERT query to add a new product to the "Products" table.
- The code uses async/await to handle asynchronous operations and database queries.
- Error handling is implemented using try-catch blocks, allowing for better error reporting.
- The code follows a modular structure, with separate files for controllers, routes, and services.

Good work on taking advantage of JavaScript and Node.js features to build a robust and efficient API! Keep it up!

**Review by denise-techlead — APPROVED** [review]
Everything seems to be fine over here. Congratulations!

**Comment by denise-techlead** [comment]
The unit tests did not pass. This means there is something wrong with your solution. Fix it and create a new commit on this branch to update this Pull Request. 👀

**Pull request merged**